Brokerages/IOrderBook.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Represents an order book interface for a security.
+    /// </summary>
+    /// <typeparam name="K">Price level identifier</typeparam>
+    /// <typeparam name="V">Size at the price level</typeparam>
+    public interface IOrderBook<K, V>
```

Comment:
Awesome, the generic arguments are required to add support the Bitmex order book in the future 👍 

I think this interface should be renamed to `IOrderBookUpdater<K, V>` since it only has methods for updating the order book data. 
The non-generic `IOrderBook` interface can be added later on when addressing issue #997.